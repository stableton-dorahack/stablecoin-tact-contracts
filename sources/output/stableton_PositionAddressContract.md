# TACT Compilation Report
Contract: PositionAddressContract
BOC Size: 572 bytes

# Types
Total Types: 19

## StateInit
TLB: `_ code:^cell data:^cell = StateInit`
Signature: `StateInit{code:^cell,data:^cell}`

## Context
TLB: `_ bounced:bool sender:address value:int257 raw:^slice = Context`
Signature: `Context{bounced:bool,sender:address,value:int257,raw:^slice}`

## SendParameters
TLB: `_ bounce:bool to:address value:int257 mode:int257 body:Maybe ^cell code:Maybe ^cell data:Maybe ^cell = SendParameters`
Signature: `SendParameters{bounce:bool,to:address,value:int257,mode:int257,body:Maybe ^cell,code:Maybe ^cell,data:Maybe ^cell}`

## Deploy
TLB: `deploy#946a98b6 queryId:uint64 = Deploy`
Signature: `Deploy{queryId:uint64}`

## DeployOk
TLB: `deploy_ok#aff90f57 queryId:uint64 = DeployOk`
Signature: `DeployOk{queryId:uint64}`

## PositionState
TLB: `_ collateral:coins debt:uint64 = PositionState`
Signature: `PositionState{collateral:coins,debt:uint64}`

## StablecoinBurnedMessage
TLB: `stablecoin_burned_message#d2be4c6c user:address amount:uint64 fees:uint64 = StablecoinBurnedMessage`
Signature: `StablecoinBurnedMessage{user:address,amount:uint64,fees:uint64}`

## DepositCollateralMessage
TLB: `deposit_collateral_message#11dfb138 user:address amount:coins = DepositCollateralMessage`
Signature: `DepositCollateralMessage{user:address,amount:coins}`

## WithdrawCollateralMessage
TLB: `withdraw_collateral_message#6829d55e user:address amount:coins debtRate:int257 = WithdrawCollateralMessage`
Signature: `WithdrawCollateralMessage{user:address,amount:coins,debtRate:int257}`

## WithdrawStablecoinMessage
TLB: `withdraw_stablecoin_message#f3554cd7 user:address amount:uint64 debtRate:uint64 = WithdrawStablecoinMessage`
Signature: `WithdrawStablecoinMessage{user:address,amount:uint64,debtRate:uint64}`

## RepayStablecoinMessage
TLB: `repay_stablecoin_message#6e81f446 user:address amount:uint64 debtRate:uint64 = RepayStablecoinMessage`
Signature: `RepayStablecoinMessage{user:address,amount:uint64,debtRate:uint64}`

## DoWithdrawCollateralMessage
TLB: `do_withdraw_collateral_message#c3cd33bf user:address amount:uint64 = DoWithdrawCollateralMessage`
Signature: `DoWithdrawCollateralMessage{user:address,amount:uint64}`

## IncreaseTotalStableMessage
TLB: `increase_total_stable_message#96096519 user:address amount:uint64 = IncreaseTotalStableMessage`
Signature: `IncreaseTotalStableMessage{user:address,amount:uint64}`

## DecreaseTotalStableMessage
TLB: `decrease_total_stable_message#ed81304a user:address amount:uint64 fees:uint64 = DecreaseTotalStableMessage`
Signature: `DecreaseTotalStableMessage{user:address,amount:uint64,fees:uint64}`

## MintFeesMessage
TLB: `mint_fees_message#9d42541d to:address amount:coins = MintFeesMessage`
Signature: `MintFeesMessage{to:address,amount:coins}`

## JettonData
TLB: `_ totalSupply:int257 mintable:bool owner:address content:Maybe ^cell walletCode:^cell = JettonData`
Signature: `JettonData{totalSupply:int257,mintable:bool,owner:address,content:Maybe ^cell,walletCode:^cell}`

## JettonWalletData
TLB: `_ balance:int257 owner:address master:address walletCode:^cell = JettonWalletData`
Signature: `JettonWalletData{balance:int257,owner:address,master:address,walletCode:^cell}`

## SetPositionIdMessage
TLB: `set_position_id_message#00b3cad5 positionId:uint8 = SetPositionIdMessage`
Signature: `SetPositionIdMessage{positionId:uint8}`

## SetPositionAddressMessage
TLB: `set_position_address_message#cf891f9e user:address = SetPositionAddressMessage`
Signature: `SetPositionAddressMessage{user:address}`

# Get Methods
Total Get Methods: 2

## getPositionId

## getPositionAddress

# Error Codes
2: Stack undeflow
3: Stack overflow
4: Integer overflow
5: Integer out of expected range
6: Invalid opcode
7: Type check error
8: Cell overflow
9: Cell underflow
10: Dictionary error
13: Out of gas error
32: Method ID not found
34: Action is invalid or not supported
37: Not enough TON
38: Not enough extra-currencies
128: Null reference exception
129: Invalid serialization prefix
130: Invalid incoming message
131: Constraints error
132: Access denied
133: Contract stopped
134: Invalid argument
135: Code of a contract was not found
136: Invalid address
4429: Invalid sender