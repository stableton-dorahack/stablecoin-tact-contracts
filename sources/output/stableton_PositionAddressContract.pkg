{"name":"PositionAddressContract","code":"te6ccgECDwEAAjAAART/APSkE/S88sgLAQIBYgIDAqDQAdDTAwFxsMABkX+RcOIB+kAiUFVvBPhh7UTQ1AH4YtIAAZjTH/pAARJsEo6KgQEB1wABAdHbPOJa2zwwyPhCAcx/AcoAWQLLHwHPFsntVA0EAgEgCAkB0HAh10nCH5UwINcLH94Cklt/4CGCEM+JH566jhwx0x8BghDPiR+euvLggfpAATFb+EFvJBAjXwN/4AGCEJRqmLa6jqLTHwGCEJRqmLa68uCB0z8BMcgBghCv+Q9XWMsfyz/J2zx/4DBwBQEm+EFvJBAjXwN/cFADgEIBbW3bPAYB9shxAcoBUAcBygBwAcoCUAXPFlAD+gJwAcpoI26zJW6zsY5MfwHKAMhwAcoAcAHKACRus51/AcoABCBu8tCAUATMljQDcAHKAOIkbrOdfwHKAAQgbvLQgFAEzJY0A3ABygDicAHKAAJ/AcoAAslYzJczMwFwAcoA4iFuswcAMJx/AcoAASBu8tCAAcyVMXABygDiyQH7AAJJvtyvaiaGoA/DFpAADMaY/9IACJNglHRUCAgOuAAIDo7Z5xbZ5A0KAgEgCwwAAjEAcbu9GCcFzsPV0srnsehOw51kqFG2aCcJ3WNS0rZHyzItOvLf3xYjmCcCBVwBuAZ2OUzlg6rkclssOAJJuoy+1E0NQB+GLSAAGY0x/6QAESbBKOioEBAdcAAQHR2zzi2zyA0OACRwIMhyAcsBcAHLABLKB8v/ydAAAjA=","abi":"{\"name\":\"PositionAddressContract\",\"types\":[{\"name\":\"StateInit\",\"header\":null,\"fields\":[{\"name\":\"code\",\"type\":{\"kind\":\"simple\",\"type\":\"cell\",\"optional\":false}},{\"name\":\"data\",\"type\":{\"kind\":\"simple\",\"type\":\"cell\",\"optional\":false}}]},{\"name\":\"Context\",\"header\":null,\"fields\":[{\"name\":\"bounced\",\"type\":{\"kind\":\"simple\",\"type\":\"bool\",\"optional\":false}},{\"name\":\"sender\",\"type\":{\"kind\":\"simple\",\"type\":\"address\",\"optional\":false}},{\"name\":\"value\",\"type\":{\"kind\":\"simple\",\"type\":\"int\",\"optional\":false,\"format\":257}},{\"name\":\"raw\",\"type\":{\"kind\":\"simple\",\"type\":\"slice\",\"optional\":false}}]},{\"name\":\"SendParameters\",\"header\":null,\"fields\":[{\"name\":\"bounce\",\"type\":{\"kind\":\"simple\",\"type\":\"bool\",\"optional\":false}},{\"name\":\"to\",\"type\":{\"kind\":\"simple\",\"type\":\"address\",\"optional\":false}},{\"name\":\"value\",\"type\":{\"kind\":\"simple\",\"type\":\"int\",\"optional\":false,\"format\":257}},{\"name\":\"mode\",\"type\":{\"kind\":\"simple\",\"type\":\"int\",\"optional\":false,\"format\":257}},{\"name\":\"body\",\"type\":{\"kind\":\"simple\",\"type\":\"cell\",\"optional\":true}},{\"name\":\"code\",\"type\":{\"kind\":\"simple\",\"type\":\"cell\",\"optional\":true}},{\"name\":\"data\",\"type\":{\"kind\":\"simple\",\"type\":\"cell\",\"optional\":true}}]},{\"name\":\"Deploy\",\"header\":2490013878,\"fields\":[{\"name\":\"queryId\",\"type\":{\"kind\":\"simple\",\"type\":\"uint\",\"optional\":false,\"format\":64}}]},{\"name\":\"DeployOk\",\"header\":2952335191,\"fields\":[{\"name\":\"queryId\",\"type\":{\"kind\":\"simple\",\"type\":\"uint\",\"optional\":false,\"format\":64}}]},{\"name\":\"PositionState\",\"header\":null,\"fields\":[{\"name\":\"collateral\",\"type\":{\"kind\":\"simple\",\"type\":\"uint\",\"optional\":false,\"format\":\"coins\"}},{\"name\":\"debt\",\"type\":{\"kind\":\"simple\",\"type\":\"uint\",\"optional\":false,\"format\":64}}]},{\"name\":\"StablecoinBurnedMessage\",\"header\":3535686764,\"fields\":[{\"name\":\"user\",\"type\":{\"kind\":\"simple\",\"type\":\"address\",\"optional\":false}},{\"name\":\"amount\",\"type\":{\"kind\":\"simple\",\"type\":\"uint\",\"optional\":false,\"format\":64}},{\"name\":\"fees\",\"type\":{\"kind\":\"simple\",\"type\":\"uint\",\"optional\":false,\"format\":64}}]},{\"name\":\"DepositCollateralMessage\",\"header\":299872568,\"fields\":[{\"name\":\"user\",\"type\":{\"kind\":\"simple\",\"type\":\"address\",\"optional\":false}},{\"name\":\"amount\",\"type\":{\"kind\":\"simple\",\"type\":\"uint\",\"optional\":false,\"format\":\"coins\"}}]},{\"name\":\"WithdrawCollateralMessage\",\"header\":1747572062,\"fields\":[{\"name\":\"user\",\"type\":{\"kind\":\"simple\",\"type\":\"address\",\"optional\":false}},{\"name\":\"amount\",\"type\":{\"kind\":\"simple\",\"type\":\"uint\",\"optional\":false,\"format\":\"coins\"}},{\"name\":\"debtRate\",\"type\":{\"kind\":\"simple\",\"type\":\"int\",\"optional\":false,\"format\":257}}]},{\"name\":\"WithdrawStablecoinMessage\",\"header\":4082453719,\"fields\":[{\"name\":\"user\",\"type\":{\"kind\":\"simple\",\"type\":\"address\",\"optional\":false}},{\"name\":\"amount\",\"type\":{\"kind\":\"simple\",\"type\":\"uint\",\"optional\":false,\"format\":64}},{\"name\":\"debtRate\",\"type\":{\"kind\":\"simple\",\"type\":\"uint\",\"optional\":false,\"format\":64}}]},{\"name\":\"RepayStablecoinMessage\",\"header\":1854010438,\"fields\":[{\"name\":\"user\",\"type\":{\"kind\":\"simple\",\"type\":\"address\",\"optional\":false}},{\"name\":\"amount\",\"type\":{\"kind\":\"simple\",\"type\":\"uint\",\"optional\":false,\"format\":64}},{\"name\":\"debtRate\",\"type\":{\"kind\":\"simple\",\"type\":\"uint\",\"optional\":false,\"format\":64}}]},{\"name\":\"DoWithdrawCollateralMessage\",\"header\":3285005247,\"fields\":[{\"name\":\"user\",\"type\":{\"kind\":\"simple\",\"type\":\"address\",\"optional\":false}},{\"name\":\"amount\",\"type\":{\"kind\":\"simple\",\"type\":\"uint\",\"optional\":false,\"format\":64}}]},{\"name\":\"IncreaseTotalStableMessage\",\"header\":2517198105,\"fields\":[{\"name\":\"user\",\"type\":{\"kind\":\"simple\",\"type\":\"address\",\"optional\":false}},{\"name\":\"amount\",\"type\":{\"kind\":\"simple\",\"type\":\"uint\",\"optional\":false,\"format\":64}}]},{\"name\":\"DecreaseTotalStableMessage\",\"header\":3984666698,\"fields\":[{\"name\":\"user\",\"type\":{\"kind\":\"simple\",\"type\":\"address\",\"optional\":false}},{\"name\":\"amount\",\"type\":{\"kind\":\"simple\",\"type\":\"uint\",\"optional\":false,\"format\":64}},{\"name\":\"fees\",\"type\":{\"kind\":\"simple\",\"type\":\"uint\",\"optional\":false,\"format\":64}}]},{\"name\":\"MintFeesMessage\",\"header\":2638369821,\"fields\":[{\"name\":\"to\",\"type\":{\"kind\":\"simple\",\"type\":\"address\",\"optional\":false}},{\"name\":\"amount\",\"type\":{\"kind\":\"simple\",\"type\":\"uint\",\"optional\":false,\"format\":\"coins\"}}]},{\"name\":\"JettonData\",\"header\":null,\"fields\":[{\"name\":\"totalSupply\",\"type\":{\"kind\":\"simple\",\"type\":\"int\",\"optional\":false,\"format\":257}},{\"name\":\"mintable\",\"type\":{\"kind\":\"simple\",\"type\":\"bool\",\"optional\":false}},{\"name\":\"owner\",\"type\":{\"kind\":\"simple\",\"type\":\"address\",\"optional\":false}},{\"name\":\"content\",\"type\":{\"kind\":\"simple\",\"type\":\"cell\",\"optional\":true}},{\"name\":\"walletCode\",\"type\":{\"kind\":\"simple\",\"type\":\"cell\",\"optional\":false}}]},{\"name\":\"JettonWalletData\",\"header\":null,\"fields\":[{\"name\":\"balance\",\"type\":{\"kind\":\"simple\",\"type\":\"int\",\"optional\":false,\"format\":257}},{\"name\":\"owner\",\"type\":{\"kind\":\"simple\",\"type\":\"address\",\"optional\":false}},{\"name\":\"master\",\"type\":{\"kind\":\"simple\",\"type\":\"address\",\"optional\":false}},{\"name\":\"walletCode\",\"type\":{\"kind\":\"simple\",\"type\":\"cell\",\"optional\":false}}]},{\"name\":\"SetPositionIdMessage\",\"header\":11782869,\"fields\":[{\"name\":\"positionId\",\"type\":{\"kind\":\"simple\",\"type\":\"uint\",\"optional\":false,\"format\":8}}]},{\"name\":\"SetPositionAddressMessage\",\"header\":3481870238,\"fields\":[{\"name\":\"user\",\"type\":{\"kind\":\"simple\",\"type\":\"address\",\"optional\":false}}]}],\"receivers\":[{\"receiver\":\"internal\",\"message\":{\"kind\":\"typed\",\"type\":\"SetPositionAddressMessage\"}},{\"receiver\":\"internal\",\"message\":{\"kind\":\"typed\",\"type\":\"Deploy\"}}],\"getters\":[{\"name\":\"getPositionId\",\"arguments\":[],\"returnType\":{\"kind\":\"simple\",\"type\":\"int\",\"optional\":false,\"format\":257}},{\"name\":\"getPositionAddress\",\"arguments\":[],\"returnType\":{\"kind\":\"simple\",\"type\":\"address\",\"optional\":false}}],\"errors\":{\"2\":{\"message\":\"Stack undeflow\"},\"3\":{\"message\":\"Stack overflow\"},\"4\":{\"message\":\"Integer overflow\"},\"5\":{\"message\":\"Integer out of expected range\"},\"6\":{\"message\":\"Invalid opcode\"},\"7\":{\"message\":\"Type check error\"},\"8\":{\"message\":\"Cell overflow\"},\"9\":{\"message\":\"Cell underflow\"},\"10\":{\"message\":\"Dictionary error\"},\"13\":{\"message\":\"Out of gas error\"},\"32\":{\"message\":\"Method ID not found\"},\"34\":{\"message\":\"Action is invalid or not supported\"},\"37\":{\"message\":\"Not enough TON\"},\"38\":{\"message\":\"Not enough extra-currencies\"},\"128\":{\"message\":\"Null reference exception\"},\"129\":{\"message\":\"Invalid serialization prefix\"},\"130\":{\"message\":\"Invalid incoming message\"},\"131\":{\"message\":\"Constraints error\"},\"132\":{\"message\":\"Access denied\"},\"133\":{\"message\":\"Contract stopped\"},\"134\":{\"message\":\"Invalid argument\"},\"135\":{\"message\":\"Code of a contract was not found\"},\"136\":{\"message\":\"Invalid address\"},\"4429\":{\"message\":\"Invalid sender\"}}}","init":{"args":[{"name":"id","type":{"kind":"simple","type":"int","optional":false,"format":257}}],"prefix":{"bits":1,"value":0},"deployment":{"kind":"system-cell","system":"te6cckECIgEABIAAAQHAAQIBWA4CAQW7dfgDART/APSkE/S88sgLBAIBYgsFAgEgCQYCASAXBwJJuoy+1E0NQB+GLSAAGY0x/6QAESbBKOioEBAdcAAQHR2zzi2zyA0IAAIwAkm+3K9qJoagD8MWkAAMxpj/0gAIk2CUdFQICA64AAgOjtnnFtnkDQoAAjECoNAB0NMDAXGwwAGRf5Fw4gH6QCJQVW8E+GHtRNDUAfhi0gABmNMf+kABEmwSjoqBAQHXAAEB0ds84lrbPDDI+EIBzH8BygBZAssfAc8Wye1UDQwB0HAh10nCH5UwINcLH94Cklt/4CGCEM+JH566jhwx0x8BghDPiR+euvLggfpAATFb+EFvJBAjXwN/4AGCEJRqmLa6jqLTHwGCEJRqmLa68uCB0z8BMcgBghCv+Q9XWMsfyz/J2zx/4DBwGwAkcCDIcgHLAXABywASygfL/8nQAQW4Q2gPART/APSkE/S88sgLEAIBYhgRAgFYFxICAnEVEwJYqqjtRNDUAfhi0gABjhL6QAEB+kABAdMH+gDTP1VAbBWOh/pAAQHR2zzi2zwhFAAEbDICWKlQ7UTQ1AH4YtIAAY4S+kABAfpAAQHTB/oA0z9VQGwVjof6QAEB0ds84ts8IRYABF8EAHG7vRgnBc7D1dLK57HoTsOdZKhRtmgnCd1jUtK2R8syLTry398WI5gnAgVcAbgGdjlM5YOq5HJbLDgCxNAB0NMDAXGwwAGRf5Fw4gH6QCJQVW8E+GHtRNDUAfhi0gABjhL6QAEB+kABAdMH+gDTP1VAbBWOh/pAAQHR2zziVRTbPDDI+EIBzH8BygBVQFBUzxZYzxbLB1j6Ass/ye1UIRkE8nAh10nCH5UwINcLH94Cklt/4CGCEBHfsTi6jhkx0x8BghAR37E4uvLggfpAAQH6AFlsElt/4CGCCLPK1bqOGjHTHwGCCLPK1bry4IHTBwExI8MAkTORMOJ/4CGCEPNVTNe64wIhghBugfRGuuMCIYIQ0r5MbLrjAiEgHx4aAbKCEGgp1V66jiAx0x8BghBoKdVeuvLggfpAAQH6AIEBAdcAVSBsE18Df+ABghCUapi2uo6i0x8BghCUapi2uvLggdM/ATHIAYIQr/kPV1jLH8s/yds8f+AwcBsBJvhBbyQQI18Df3BQA4BCAW1t2zwcAfbIcQHKAVAHAcoAcAHKAlAFzxZQA/oCcAHKaCNusyVus7GOTH8BygDIcAHKAHABygAkbrOdfwHKAAQgbvLQgFAEzJY0A3ABygDiJG6znX8BygAEIG7y0IBQBMyWNANwAcoA4nABygACfwHKAALJWMyXMzMBcAHKAOIhbrMdADCcfwHKAAEgbvLQgAHMlTFwAcoA4skB+wAAOjHTHwGCENK+TGy68uCB+kABAdM/0z9VIGwTXwN/ADox0x8BghBugfRGuvLggfpAAQHTP9M/VSBsE18DfwA6MdMfAYIQ81VM17ry4IH6QAEB0z/TP1UgbBNfA38ALnBUcAAgyHIBywFwAcsAEsoHy//J0FUgi9DcPg=="}},"compiler":{"name":"tact","version":"0.10.1"}}