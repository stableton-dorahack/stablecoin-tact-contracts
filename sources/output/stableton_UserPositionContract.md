# TACT Compilation Report
Contract: UserPositionContract
BOC Size: 1678 bytes

# Types
Total Types: 25

## StateInit
TLB: `_ code:^cell data:^cell = StateInit`
Signature: `StateInit{code:^cell,data:^cell}`

## Context
TLB: `_ bounced:bool sender:address value:int257 raw:^slice = Context`
Signature: `Context{bounced:bool,sender:address,value:int257,raw:^slice}`

## SendParameters
TLB: `_ bounce:bool to:address value:int257 mode:int257 body:Maybe ^cell code:Maybe ^cell data:Maybe ^cell = SendParameters`
Signature: `SendParameters{bounce:bool,to:address,value:int257,mode:int257,body:Maybe ^cell,code:Maybe ^cell,data:Maybe ^cell}`

## Deploy
TLB: `deploy#946a98b6 queryId:uint64 = Deploy`
Signature: `Deploy{queryId:uint64}`

## DeployOk
TLB: `deploy_ok#aff90f57 queryId:uint64 = DeployOk`
Signature: `DeployOk{queryId:uint64}`

## ChangeOwner
TLB: `change_owner#0f474d03 newOwner:address = ChangeOwner`
Signature: `ChangeOwner{newOwner:address}`

## DepositCollateralMessage
TLB: `deposit_collateral_message#11dfb138 user:address amount:coins = DepositCollateralMessage`
Signature: `DepositCollateralMessage{user:address,amount:coins}`

## WithdrawCollateralMessage
TLB: `withdraw_collateral_message#6829d55e user:address amount:coins debtRate:int257 = WithdrawCollateralMessage`
Signature: `WithdrawCollateralMessage{user:address,amount:coins,debtRate:int257}`

## WithdrawStablecoinMessage
TLB: `withdraw_stablecoin_message#f3554cd7 user:address amount:uint64 debtRate:uint64 = WithdrawStablecoinMessage`
Signature: `WithdrawStablecoinMessage{user:address,amount:uint64,debtRate:uint64}`

## RepayStablecoinMessage
TLB: `repay_stablecoin_message#6e81f446 user:address amount:uint64 debtRate:uint64 = RepayStablecoinMessage`
Signature: `RepayStablecoinMessage{user:address,amount:uint64,debtRate:uint64}`

## DoWithdrawCollateralMessage
TLB: `do_withdraw_collateral_message#c3cd33bf user:address amount:uint64 = DoWithdrawCollateralMessage`
Signature: `DoWithdrawCollateralMessage{user:address,amount:uint64}`

## IncreaseTotalStableMessage
TLB: `increase_total_stable_message#96096519 user:address amount:uint64 = IncreaseTotalStableMessage`
Signature: `IncreaseTotalStableMessage{user:address,amount:uint64}`

## DecreaseTotalStableMessage
TLB: `decrease_total_stable_message#ed81304a user:address amount:uint64 fees:uint64 = DecreaseTotalStableMessage`
Signature: `DecreaseTotalStableMessage{user:address,amount:uint64,fees:uint64}`

## MintFeesMessage
TLB: `mint_fees_message#9d42541d to:address amount:coins = MintFeesMessage`
Signature: `MintFeesMessage{to:address,amount:coins}`

## JettonData
TLB: `_ totalSupply:int257 mintable:bool owner:address content:Maybe ^cell walletCode:^cell = JettonData`
Signature: `JettonData{totalSupply:int257,mintable:bool,owner:address,content:Maybe ^cell,walletCode:^cell}`

## JettonWalletData
TLB: `_ balance:int257 owner:address master:address walletCode:^cell = JettonWalletData`
Signature: `JettonWalletData{balance:int257,owner:address,master:address,walletCode:^cell}`

## SetUserPositionDependecyMessage
TLB: `set_user_position_dependecy_message#f037fcfb stablecoinMasterAddress:address positionsManagerAddress:address = SetUserPositionDependecyMessage`
Signature: `SetUserPositionDependecyMessage{stablecoinMasterAddress:address,positionsManagerAddress:address}`

## SetPositionIdMessage
TLB: `set_position_id_message#4a35f9d0 user:address positionId:int257 = SetPositionIdMessage`
Signature: `SetPositionIdMessage{user:address,positionId:int257}`

## NewPositionIdMessage
TLB: `new_position_id_message#0d74c595 user:address = NewPositionIdMessage`
Signature: `NewPositionIdMessage{user:address}`

## SetPositionAddressMessage
TLB: `set_position_address_message#5da6de96 user:address position:address = SetPositionAddressMessage`
Signature: `SetPositionAddressMessage{user:address,position:address}`

## MintMessage
TLB: `mint_message#903559b2 user:address amount:coins = MintMessage`
Signature: `MintMessage{user:address,amount:coins}`

## BurnMessage
TLB: `burn_message#b625719b user:address amount:coins fees:coins = BurnMessage`
Signature: `BurnMessage{user:address,amount:coins,fees:coins}`

## RepayBurnNotification
TLB: `repay_burn_notification#563c6d94 user:address amount:coins = RepayBurnNotification`
Signature: `RepayBurnNotification{user:address,amount:coins}`

## StablecoinBurnedMessage
TLB: `stablecoin_burned_message#d2be4c6c user:address amount:uint64 fees:uint64 = StablecoinBurnedMessage`
Signature: `StablecoinBurnedMessage{user:address,amount:uint64,fees:uint64}`

## PositionState
TLB: `_ collateral:coins debt:uint64 = PositionState`
Signature: `PositionState{collateral:coins,debt:uint64}`

# Get Methods
Total Get Methods: 4

## getPositionUser

## getPositionId

## getPositionState

## owner

# Error Codes
2: Stack undeflow
3: Stack overflow
4: Integer overflow
5: Integer out of expected range
6: Invalid opcode
7: Type check error
8: Cell overflow
9: Cell underflow
10: Dictionary error
13: Out of gas error
32: Method ID not found
34: Action is invalid or not supported
37: Not enough TON
38: Not enough extra-currencies
128: Null reference exception
129: Invalid serialization prefix
130: Invalid incoming message
131: Constraints error
132: Access denied
133: Contract stopped
134: Invalid argument
135: Code of a contract was not found
136: Invalid address
15032: not from stablecoin master
22230: Already set
31797: debt less than repay amount
53160: not from positions manager
61910: not from positionsManager
63577: withdrawal amount more than position has