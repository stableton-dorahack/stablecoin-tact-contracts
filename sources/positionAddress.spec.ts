import { Address, OpenedContract, Sender, toNano } from "ton";
import { ContractSystem, Logger, Tracker, Treasure } from "ton-emulator";
import { UserPositionContract } from "./output/stableton_UserPositionContract";
import { PositionAddressContract } from "./output/stableton_PositionAddressContract";

describe("PositionAddressContract", () => {
    let system: ContractSystem;
    let owner: Treasure;
    let user: Treasure;
    let position: Treasure;
    let positionsManager: Treasure;

    let positionAddressContract: OpenedContract<PositionAddressContract>;
    let track: Tracker;
    let logger: Logger;

    beforeAll(async () => {
        system = await ContractSystem.create();
        owner = system.treasure("owner");
        user = system.treasure("user");
        position = system.treasure("position");

        positionsManager = system.treasure("positionsManager");
        positionAddressContract = system.open(await PositionAddressContract.fromInit(1n, positionsManager.address));
        track = system.track(positionAddressContract.address);
    });

    it("should deploy correctly", async () => {
        await positionAddressContract.send(owner, { value: toNano(1) }, { $$type: "Deploy", queryId: 0n });

        await system.run();
        expect(track.collect()).toMatchSnapshot();

        expect(await positionAddressContract.getGetPositionId()).toEqual(1n);
    });

    it("on setPositionAddress from userPosition", async () => {
        await positionAddressContract.send(
            owner,
            { value: toNano(101) },
            { $$type: "SetPositionAddressMessage", user: user.address, position: position.address }
        );

        await system.run();
        expect(track.collect()).toMatchSnapshot();

        expect((await positionAddressContract.getGetPositionAddress()).toString()).toEqual(position.address.toString());
    });
});
