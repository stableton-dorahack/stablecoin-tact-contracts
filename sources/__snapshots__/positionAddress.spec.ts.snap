// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PositionAddressContract on setPositionAddress from userPosition 1`] = `
[
  {
    "$seq": 1,
    "events": [
      {
        "$type": "received",
        "message": {
          "body": {
            "cell": "x{5DA6DE96800784F79BB9BDDE39E2B05A2820A591FBCD92F3ACC083685CD65E37ED00489A4B30001DC8F849F1A1B451DD1D5ED0C497DF2EB583D7FC37EFC48BD5D38F1B4073EB72_}",
            "type": "cell",
          },
          "bounce": true,
          "from": "kQAI-3FJVc_ywSuY4vq0bYrzR7S4Och4y7bTU_i5yLOB3A6P",
          "to": "kQAvYPy9IeUb7xIP3vA_R_VGhWbfxYdB-dBIrgWiESamKzhQ",
          "type": "internal",
          "value": 101000000000n,
        },
      },
      {
        "$type": "failed",
        "errorCode": 22230,
        "errorMessage": "Already set",
      },
      {
        "$type": "sent-bounced",
        "message": {
          "body": {
            "type": "empty",
          },
          "bounce": false,
          "from": "kQAvYPy9IeUb7xIP3vA_R_VGhWbfxYdB-dBIrgWiESamKzhQ",
          "to": "kQAI-3FJVc_ywSuY4vq0bYrzR7S4Och4y7bTU_i5yLOB3A6P",
          "type": "internal",
          "value": 100995858000n,
        },
      },
    ],
  },
]
`;

exports[`PositionAddressContract should deploy correctly 1`] = `
[
  {
    "$seq": 0,
    "events": [
      {
        "$type": "deploy",
      },
      {
        "$type": "received",
        "message": {
          "body": {
            "cell": "x{946A98B60000000000000000}",
            "type": "cell",
          },
          "bounce": true,
          "from": "kQAI-3FJVc_ywSuY4vq0bYrzR7S4Och4y7bTU_i5yLOB3A6P",
          "to": "kQAvYPy9IeUb7xIP3vA_R_VGhWbfxYdB-dBIrgWiESamKzhQ",
          "type": "internal",
          "value": 1000000000n,
        },
      },
      {
        "$type": "processed",
        "gasUsed": 7369n,
      },
      {
        "$type": "sent",
        "messages": [
          {
            "body": {
              "cell": "x{AFF90F570000000000000000}",
              "type": "cell",
            },
            "bounce": true,
            "from": "kQAvYPy9IeUb7xIP3vA_R_VGhWbfxYdB-dBIrgWiESamKzhQ",
            "to": "kQAI-3FJVc_ywSuY4vq0bYrzR7S4Och4y7bTU_i5yLOB3A6P",
            "type": "internal",
            "value": 991435000n,
          },
        ],
      },
    ],
  },
]
`;
